vavr/src/main/java/io/vavr/collection/Collections.java

```diff
@@ -51,7 +51,9 @@ static boolean areEqual(Iterable<?> iterable1, Iterable<?> iterable2) {
                     .foldLeft(Iterator.of(empty), (product, ignored) -> product.flatMap(el -> seq.map(t -> (S) el.append(t))));
         }
     }
-    
+
+    // DEV-NOTE: Use this method for non-infinite and direct-access collection only
+    // see https://github.com/vavr-io/vavr/issues/2007
```

Comment:
Could you rather add a little explanation why one should not use this method when using infinite collections? I think this is more convinient than referring to a github issue which might get lost over the time.